roofit/xroofit/src/xRooNode.cxx

```diff
@@ -5742,12 +6867,12 @@ class PdfWrapper : public RooAbsPdf {
       }
 
       // Clone self for internal use
-      RooAbsReal *cloneFunc = (RooAbsReal *)fFunc.absArg()->cloneTree();
-      RooAbsPdf *clonePdf = dynamic_cast<RooAbsPdf *>(cloneFunc);
-      std::unique_ptr<RooArgSet> errorParams{cloneFunc->getObservables(fpf_stripped)};
+      RooAbsReal *cloneFunc = const_cast<PdfWrapper*>(this);// (RooAbsReal *)fFunc.absArg()->cloneTree();
+      //RooAbsPdf *clonePdf = dynamic_cast<RooAbsPdf *>(cloneFunc);
+      RooArgSet *errorParams = cloneFunc->getObservables(fpf_stripped);
 
-      std::unique_ptr<RooArgSet> nset =
-         nset_in.empty() ? std::unique_ptr<RooArgSet>{cloneFunc->getParameters(*errorParams)} : std::unique_ptr<RooArgSet>{cloneFunc->getObservables(nset_in)};
+      RooArgSet *nset =
+         nset_in.getSize() == 0 ? cloneFunc->getParameters(*errorParams) : cloneFunc->getObservables(nset_in);
```

Comment:
this code is only needed for old ROOT versions, so hid it in an if block rather than modifying it. Will remove in future versions (I think I'm going to remove support for 6.26 once 6.30 is out).